llama-index-integrations/vector_stores/llama-index-vector-stores-opensearch/llama_index/vector_stores/opensearch/base.py

```diff
@@ -210,16 +213,90 @@ def _default_approximate_search_query(
             "query": {"knn": {vector_field: {"vector": query_vector, "k": k}}},
         }
 
-    def _parse_filters(self, filters: Optional[MetadataFilters]) -> Any:
-        pre_filter = []
-        if filters is not None:
-            for f in filters.legacy_filters():
-                if isinstance(f.value, str):
-                    pre_filter.append({f.key: f.value})
-                else:
-                    pre_filter.append({f.key: json.loads(str(f.value))})
+    def _is_text_field(self, value: Any) -> bool:
+        """Check if value is a string and keyword filtering needs to be performed.
 
-        return pre_filter
+        Not applied to datetime strings.
+        """
+        if isinstance(value, str):
+            try:
+                datetime.fromisoformat(value)
+                return False
+            except ValueError as e:
+                print("Not a date field: ", value, e)
```

Comment:
lets remove this print statement

Reply:
Oops, debugging leftover 🙈 Thanks for catching that